### PR TITLE
Course controller and IT test refactor

### LINKED ISSUE #50: Implement REST controller for Course

### PR DESCRIPTION
**Pull request by emmaeagles**
# Summary

ok so this is the course controller but also i made a class that holds all the test create methods for the IT tests so that we dont have the same methods in every file

## Test Plan

How did you test your changes?

the tests will all pass once  my other branch is merged and i can rebase (???)

## Related Issues

Which issues does this PR resolve/work on?
closes #50

### CODE REVIEW

File: Backend/src/test/java/ca/mcgill/cooperator/controller/StudentReportControllerIT.java [hunk 1]
```diff
@@ -38,17 +31,12 @@
 @SpringBootTest
 @AutoConfigureMockMvc
 @ActiveProfiles("test")
-public class StudentReportControllerIT {
+public class StudentReportControllerIT extends ControllerIT {
```
Comment (kraglalbert) [inline Backend/src/test/java/ca/mcgill/cooperator/controller/StudentReportControllerIT.java]: This is really smart not sure why we didn't do this earlier 😮

File: Backend/src/test/java/ca/mcgill/cooperator/controller/ControllerIT.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+import ca.mcgill.cooperator.dto.StudentDto;
+import ca.mcgill.cooperator.model.CoopStatus;
+import ca.mcgill.cooperator.model.Season;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+public class ControllerIT {
```
Comment (kraglalbert) [inline Backend/src/test/java/ca/mcgill/cooperator/controller/ControllerIT.java]: Beautiful 😭

File: Backend/src/main/java/ca/mcgill/cooperator/controller/CourseController.java [hunk 3]
```diff
@@ -17,18 +27,94 @@
...
+
+        return ControllerUtils.convertToDto(createdCourse);
+    }
+
+    /**
+     * Update an existing Course
+     *
+     * <p>In request body:
+     *
+     * @param name
```
Comment (kraglalbert) [inline Backend/src/main/java/ca/mcgill/cooperator/controller/CourseController.java]: Update the comment with all params the method needs and what gets returned

File: Backend/src/main/java/ca/mcgill/cooperator/controller/CourseController.java [hunk 4]
```diff
@@ -17,18 +27,94 @@
...
+        return ControllerUtils.convertToDto(deletedCourse);
+    }
+
+    @ExceptionHandler(RuntimeException.class)
+    public final ResponseEntity<Exception> handleAllExceptions(RuntimeException ex) {
+        return new ResponseEntity<Exception>(ex, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
 
-        return ControllerUtils.convertToDto(course);
+    private List<CourseOffering> convertCourseOfferingListToDomainObject(
```
Comment (kraglalbert) [inline Backend/src/main/java/ca/mcgill/cooperator/controller/CourseController.java]: This should probably go in ControllerUtils?

**Comment by kraglalbert** [comment]
Also if you write "closes #50" in the PR description it'll automatically close that issue when you merge the PR

File: Backend/src/main/java/ca/mcgill/cooperator/controller/CourseController.java (hunk 4 above)
Comment (emmaeagles) [reply Backend/src/main/java/ca/mcgill/cooperator/controller/CourseController.java]: @kraglalbert i had it there but it cant be static so i moved it here (((i think that was the error but i cant remember exactly but i moved it out of controllerutils on purpose)))

Comment (emmaeagles) [reply Backend/src/main/java/ca/mcgill/cooperator/controller/CourseController.java]: oh i think that it was that the course service needed in a method was null when i was testing even though it was autowired

**Pull request merged**